### PR TITLE
Basic auth protection on routes

### PR DESCRIPTION
**Pull request by JParisFerrer**
- Redirects most GETs to /login
- 403 unauthed POSTs
- add an auth helper for being course owner
- Validate course deletions, course instance creations are done by an owner
- creating a course, courseinstance now gives you rights to that course/courseinstance
- remove the showcase code from the list of instances
- fixed empty row when no course instances
- fixed everything being a link to logout due to unclosed <a> in the header
- add UNIQUE to all serial

Possibly missed some routes/perms.

Definite todo: add UI for adding/removing other users to a course/courseinstance as owner/staff.

### CODE REVIEW

File: src/pages/course/course.js [hunk 1]
```diff
@@ -25,28 +31,29 @@ router.get(
...
-    // showcasing using the checks
-    if (await checks.isLoggedIn(req)) {
-      res.locals.test_perms = await (async () => {
-        return Promise.all(
-          result.rows.map(r =>
-            checks.staffHasPermissionsForCourseInstance(req, r.id)
-          )
-        )
-      })()
+    if (result.rows.length >= 1 && courseRow.name !== undefined && courseRow.name !== null) {
```
Comment (JParisFerrer) [inline src/pages/course/course.js:34]: this works™ but didn't really think about a minimal check

**Comment by Binary-Eater** [comment]
"Looks good to me"

**Comment by Binary-Eater** [comment]
Will actually review shortly.

**Review by Binary-Eater — APPROVED** [review]

**Comment by Binary-Eater** [comment]
18 hours = shortly

**Pull request merged**